compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/FirHelpers.kt

```diff
@@ -105,4 +116,43 @@ fun ConeKotlinType.toRegularClass(session: FirSession): FirRegularClass? {
  */
 fun FirTypeRef.toRegularClass(session: FirSession): FirRegularClass? {
     return safeAs<FirResolvedTypeRef>()?.type?.toRegularClass(session)
+}
+
+/**
+ * Returns the ClassLikeDeclaration where the function has been defined
+ * or null if no proper declaration has been found.
+ */
+fun FirFunctionCall.getClassLikeDeclaration(context: CheckerContext): FirClassLikeDeclaration<*>? {
```

Comment:
Here I'd prefer name like `getCalledFunctionContainingClass`

Reply:
All right, I'll change it